ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java

```diff
@@ -335,9 +335,7 @@ static boolean isConnectErrorResponse(final JsonObject ksqlEntity) {
           source.getString("valueFormat"),
           formatQueries(source.getJsonArray("readQueries")),
           formatQueries(source.getJsonArray("writeQueries")),
-          source.getString("timestamp").isEmpty()
-              ? Optional.empty()
-              : Optional.of(source.getString("timestamp")),
+          Optional.ofNullable(source.getString("timestamp")),
```

Comment:
side cleanup

Reply:
Hmmm... Good catch. I think I mixed up `null` and empty string... (Guess it's a case of a missing test that should have guarded this change if we want to treat them differently...; or it does really not matter and we don't want to treat them differently). Maybe @agavra knows?

Reply:
this is public API, so we should err against changing it either way - @vcrfxia should know best, but it sounds to me like we want to treat empty as `null` (when would an empty string timestamp make sense?)

Reply:
Almog is correct -- empty string should translate to `Optional.empty()`, though there's a unit test that covers this case: https://github.com/confluentinc/ksql/blob/master/ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/ClientTest.java#L1415 so if you're saying it didn't fail with the change above, then the rest of the translation pipeline must be agnostic to it somewhere.

Reply:
Follow up PR: https://github.com/confluentinc/ksql/pull/7475